test/test_testing.py

```diff
@@ -5854,3 +5854,34 @@ def test_get_notices(self):
 class TestNotices(unittest.TestCase, _TestingPebbleClientMixin, PebbleNoticesMixin):
     def setUp(self):
         self.client = self.get_testing_client()
+
+
+class TestCloudSpec(unittest.TestCase):
+    def test_get_set_cloud_spec(self):
+        harness = ops.testing.Harness(EventRecorder, meta='name: myapp')
+        self.addCleanup(harness.cleanup)
+        cloud_spec_dict = {
+            'name': 'localhost',
+            'type': 'lxd',
+            'endpoint': 'https://127.0.0.1:8443',
+            'credential': {
+                'authtype': 'certificate',
+                'attrs': {
+                    'client-cert': 'foo',
+                    'client-key': 'bar',
+                    'server-cert': 'baz'
+                },
+            },
+        }
+        harness.set_cloud_spec(ops.model.CloudSpec.from_dict(cloud_spec_dict))
+        harness.begin()
+        result = harness.model.get_cloud_spec()
+        expected = ops.model.CloudSpec.from_dict(cloud_spec_dict)
+        self.assertEqual(repr(result), repr(expected))
```

Comment:
Note that we should be asserting on the `repr` here, but the actual object (dataclasses define equality).